### PR TITLE
upgrade to protocol v3 due to #375 PR

### PR DESCRIPTION
**Pull request by yevgenypats**
https://github.com/cloudquery/cloudquery/pull/375

### CODE REVIEW

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ func newRemotePlugin(details *registry.ProviderDetails, alias string, env []stri
 	client := plugin.NewClient(&plugin.ClientConfig{
 		HandshakeConfig: serve.Handshake,
 		VersionedPlugins: map[int]plugin.PluginSet{
-			2: pluginMap,
+			3: pluginMap,
```
Comment (roneli) [inline pkg/plugin/plugin.go:49]:
```suggestion
			3: pluginMap,
			2: pluginMap,
```

You need to support also previous protocol, I think the above should do the trick

**Review by roneli — CHANGES_REQUESTED** [review]

File: pkg/plugin/plugin.go (hunk 1 above)
Comment (yevgenypats) [reply pkg/plugin/plugin.go:49]: This wont work here as we will need to get the json back and we removed this code. This is why we release first the providers and then cloudquery core. so users will be able to upgrade to latest provider

File: pkg/plugin/plugin.go [hunk 2]
```diff
@@ -55,6 +56,10 @@ func newRemotePlugin(details *registry.ProviderDetails, alias string, env []stri
 	rpcClient, err := client.Client()
 	if err != nil {
 		client.Kill()
+		// give a more clear message to the user
+		if strings.Contains(err.Error(), "Incompatible API version") {
```
Comment (disq) [inline pkg/plugin/plugin.go:60]: Are you sure the error comes in `client.Client()` and not few lines below in `Dispense()`?

Comment (yevgenypats) [reply pkg/plugin/plugin.go:60]: yeah. tested it.

**Review by roneli — APPROVED** [review]

**Pull request merged**